Libraries/LibWeb/SVG/SVGScriptElement.cpp

```diff
@@ -40,33 +42,52 @@ void SVGScriptElement::process_the_script_element()
     if (m_already_processed || !in_a_document_tree())
         return;
 
-    auto inline_script = child_text_content();
+    String script_content;
+    auto script_url = m_document->url();
 
-    // FIXME: 2. If the 'script' element references external script content, then the external script content
-    //           using the current value of the 'xlink:href' attribute is fetched. Further processing of the
-    //           'script' element is dependent on the external script content, and will block here until the
-    //           resource has been fetched or is determined to be an invalid IRI reference.
+    // 2. If the 'script' element references external script content, then the external script content
+    //    using the current value of the 'xlink:href' attribute is fetched. Further processing of the
+    //    'script' element is dependent on the external script content, and will block here until the
+    //    resource has been fetched or is determined to be an invalid IRI reference.
     if (has_attribute(SVG::AttributeNames::href) || has_attribute_ns(Namespace::XLink.to_string(), SVG::AttributeNames::href)) {
-        dbgln("FIXME: Unsupported external fetch of SVGScriptElement!");
-        return;
+        auto href_value = href()->base_val();
+
+        script_url = document().parse_url(href_value);
+        if (!script_url.is_valid()) {
+            // Handle invalid IRI reference
+            dbgln("Invalid script URL: {}", href_value);
+            return;
+        }
+
+        // Fetch the external script content
+        auto request = LoadRequest::create_for_url_on_page(script_url, &document().page());
+        auto resource = ResourceLoader::the().load_resource(Resource::Type::Generic, request);
```

Comment:
This should be updated to use fetch, see: https://github.com/LadybirdBrowser/ladybird/issues/2634

Reply:
Sounds like a right thing to do, but looks overcomplicated. I haven't found a simple few-lines way to do that (yet)

Reply:
@shannonbooth it uses fetch now